web3c/confidential_provider.js

```diff
@@ -84,6 +88,9 @@ class ConfidentialSendTransform {
     if (!tx.to) {
       // deploy transaction doesn't encrypt anything for v0.5
       tx.data = this._prependConfidential(tx.data);
+      if (tx.header) {
```

Comment:
Note that for the first iteration of this, the plan is to **not** rip out the confidential prefix in the transaction data, so that we can minimize runtime changes. So we will have both a confidential prefix and the header option.

Reply:
should we have a set of known header keys we allow to be serialized?

Reply:
Keys for whom?

Reply:
for key-value pairs. having the library white-list known-recognized keys that will impact the contract, and not allow arbitrary data to be set there.

Reply:
Yes agreed. Will wait for the runtime changes to finalize to do this. Adding a todo.